src/test/java/core/basesyntax/operations/PurchaseOperationTest.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.operations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.operations.exception.InvalidInputDataException;
+import core.basesyntax.service.ReportCreator;
+import core.basesyntax.service.impl.ReportCreatorImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseOperationTest {
+    private PurchaseOperation purchaseOperation;
+    private ReportCreator reportCreator;
+
+    @BeforeEach
+    void setUp() {
+        reportCreator = new ReportCreatorImpl();
+        purchaseOperation = new PurchaseOperation();
+    }
+
+    @Test
+    void handle_purchaseOperation_Ok() {
+        Storage.storage.put("banana", 30);
+        String expected = "fruit,quantity"
+                + System.lineSeparator()
+                + "banana,10";
+        purchaseOperation.handle("banana", 20);
+        String result = reportCreator.createReport();
+        assertEquals(expected, result);
```

Comment:
better to make test in this way: 1.Precondition 2. Executioin 3. Assertions. And each step separate by empty line
```suggestion
        Storage.storage.put("banana", 30);
        String expected = "fruit,quantity"
                + System.lineSeparator()
                + "banana,10";
                
        purchaseOperation.handle("banana", 20);
        String result = reportCreator.createReport();
        
        assertEquals(expected, result);
```